x-pack/plugins/gis/public/selectors/map_selectors.js

```diff
@@ -93,6 +93,10 @@ const getSelectedLayerId = ({ map }) => {
 
 export const getLayerListRaw = ({ map }) => map.layerList ?  map.layerList : [];
 
+export const getWaitingForMapReadyLayerListRaw = ({ map }) => map.waitingForMapReadyLayerList
+  ? map.waitingForMapReadyLayerList
```

Comment:
Just curious, were you hitting a situation where this condition was necessary?

Reply:
yes. When you first load a gis-map saved object. The saved object state is loaded into the store but the GisApp component is not mounted so the map extent is not known. `getWaitingForMapReadyLayerListRaw ` could return layers if the map contains layers or its easier to return an empty list of the map does not contain layers and then have to check for undefined later